internal/mbsc/mbsc.go

```diff
@@ -5,38 +5,35 @@ import (
 	"fmt"
 	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
 
+	v1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 //go:generate mockgen -source=mbsc.go -package=mbsc -destination=mock_mbsc.go
 
 type MBSC interface {
-	SetModuleImageSpec(mbscObj *kmmv1beta1.ModuleBuildSignConfig, moduleImageSpec *kmmv1beta1.ModuleImageSpec)
 	GetMBSC(ctx context.Context, name, namespace string) (*kmmv1beta1.ModuleBuildSignConfig, error)
+	PatchMBSC(ctx context.Context, name string, namespace string, moduleImageSpec *kmmv1beta1.ModuleImageSpec,
```

Comment:
WDYT about renaming to `ApplyMBSC`? Sometime this will create a new object and not just patch. Also it will be more consistent with the MIC package.

Reply:
it does not apply the whole object, as in case with MBSC

Reply:
I think that Patch is more explanatory, since it means that not the whole object is updated, but only a part of it

Reply:
But it is possible that the whole object is being created since you are calling `CreateOrPatch`. If we wan't to patch only then why don't you use `client.Patch()`?

Reply:
The object will be created once, but after that every time there is only a Patch

Reply:
So the function is Creating or Patching :) not just patching. Someone looking at the function call may think no new object will be created when a Patch function is called.

Reply:
How about CreateOrPatch

Reply:
I am OK with that, we if we choose `CreateOrPatch` can you please update it in the `MIC` package as well to be consistent?

Reply:
ok, but in a different PR